sdk/eventhub/Azure.Messaging.EventHubs/tests/Producer/EventHubProducerClientTests.cs

```diff
@@ -645,6 +646,525 @@ public async Task CloseAsyncDoesNotCloseTheConnectionWhenNotOwned()
+
+            Assert.That(async () => await producerClient.SendAsync(events, options), Throws.InstanceOf<EventHubsException>().And.Property(nameof(EventHubsException.Reason)).EqualTo(EventHubsException.FailureReason.ClientClosed));
+
+            transportProducer.Verify(t => t.SendAsync(It.IsAny<IEnumerable<EventData>>(),
+                                                      It.IsAny<SendEventOptions>(),
+                                                      It.IsAny<CancellationToken>()),
+                                     Times.Once);
+        }
+
+        /// <summary>
+        ///   Retry logic starts only when the <see cref="EventHubConnection"/> owned by a <see cref="EventHubProducerClient"/> is open.
+        /// </summary>
+        ///
+        [Test]
+        public async Task RetryLogicWorksForOpenConnectionsBatch()
+        {
+            var batchOptions = new CreateBatchOptions { PartitionId = "0" };
+            var batch = new EventDataBatch(new MockTransportBatch(), batchOptions.ToSendOptions());
+            var transportProducer = new Mock<TransportProducer>();
+            var eventHubConnection = new MockConnection(() => transportProducer.Object);
+            var retryPolicy = new EventHubProducerClientOptions().RetryOptions.ToRetryPolicy();
+            var mockTransportProducerPool = new MockTransportProducerPool(transportProducer.Object, eventHubConnection, retryPolicy);
+            var mockPooledProducer = mockTransportProducerPool.GetPooledProducer(batchOptions.PartitionId) as MockPooledProducer;
+            var producerClient = new EventHubProducerClient(eventHubConnection, transportProducer.Object, mockTransportProducerPool);
+
+            transportProducer
+                .Setup(transportProducer => transportProducer.SendAsync(It.IsAny<EventDataBatch>(),
+                                                                        It.IsAny<CancellationToken>()))
+                .Throws(new EventHubsException(false, "test", EventHubsException.FailureReason.ClientClosed))
+                .Verifiable();
+
+            transportProducer
+                .SetupGet(transportProducer => transportProducer.IsClosed)
+                .Returns(true);
+
+            await eventHubConnection.CloseAsync(CancellationToken.None);
+
+            Assert.That(async () => await producerClient.SendAsync(batch), Throws.InstanceOf<EventHubsException>().And.Property(nameof(EventHubsException.Reason)).EqualTo(EventHubsException.FailureReason.ClientClosed));
+
+            transportProducer.Verify(t => t.SendAsync(It.IsAny<EventDataBatch>(),
+                                                      It.IsAny<CancellationToken>()),
+                                     Times.Once);
+        }
+
+        /// <summary>
+        ///   Retry logic starts only when the a <see cref="EventHubProducerClient"/> is open.
+        /// </summary>
+        ///
+        [Test]
+        public void RetryLogicWorksForOpenEventHubProducerClients()
```

Comment:
Again, I would suggest renaming this and the batch one to `RetryLogicDoesNotWorkForClosedEventHubProducerClients`.